src/Debugger/Impl/DebugSession.cs

```diff
@@ -143,8 +143,9 @@ public sealed class DebugSession : IDisposable {
             using (var eval = await RSession.BeginEvaluationAsync(cancellationToken)) {
                 res = await eval.EvaluateAsync(expression, json ? REvaluationKind.Json : REvaluationKind.Normal);
                 if (res.ParseStatus != RParseStatus.OK || res.Error != null || (json && res.JsonResult == null)) {
-                    Trace.Fail(Invariant($"Internal debugger evaluation {expression} failed: {res}"));
-                    throw new REvaluationException(res);
+                    string message = Invariant($"Internal debugger evaluation {expression} failed: {res}");
+                    Trace.Fail(message);
```

Comment:
In the debugger, kinda, because it's a COM API, so all exceptions get translated to HRESULTs at the AD7 boundary (and more often than not, swallowed silently by VS). 

Ideally, these asserts should rather be in AD7 code. When I get the time to go do some cleanup there, I'm going to do a generic wrapper, and just use it in all methods. But that's a much bigger change.